### ISSUE TITLE
Reihenfolge der MZP falsch rum

### ISSUE THREAD

**Issue by SonjaKaufmann** [Optinomic/apps#95]
http://optinomic.cust.local/api/#/patient/824/stay/684/app/ch.suedhang.apps.bscl_anq.production

Bin Mal wieder auf einen komischen Fall gestossen, der Patient hat beim letzten Aufenthalt die Austrittswerte vor den Eintrittswerten dargestellt, obwohl das Datum damit absteigend ist. Normalerweise wird doch erst der Eintritt dargestellt und dann der Austritt. Verstehe dies nicht, es scheint auch nicht der richtige Stay mit dem Fall verknüpft zu sein. 
Schaut den doch Mal an.

**Comment by schoenenb** [comment]
Ich schreib mir die auf. Brauchst du das für einen Export? Sonst würde ich's eher hinten anhängen.

**Comment by schoenenb** [comment]
http://optinomic.cust.local/api/#/patient/824/stay/588/app/ch.suedhang.apps.bscl_anq.production
Nach wie vor falsch rum - ev. wegen Sortierung nach datestamp statt Variable "Datum"?

**Comment by schoenenb** [comment]
Du hast zwar in #75 geschrieben

> Done Datum = q504V00: Innerhalb App & PDF

...sortieren tut es aber immer noch nach datestamp oder startdate oder so. Im folgenden Fall sagt q504V00 zwar, dass der Eintritt vor dem Übertritt ausgefüllt wurde. Da der Eintritts-BSCL jedoch einen Tag nach dem Übertritts-BSCL erfasst wurde, wird er in der Abfolge unterhalb ausgegeben, was halt verwirrt:
http://optinomic.cust.local/api/#/patient/2940/stay/1363/app/ch.suedhang.apps.bscl_anq.production

**Comment by schoenenb** [comment]
Dazu eine weitere Frage:
Worauf bezieht sich "sr.aasse.zeipunkt_datum | date"?
**GitHub\apps\ch.suedhang.apps.aase-g\templates\score_scales.html**
Z8  `            <h3 class="md-subhead">{{sr.aase.zeipunkt_datum | date}}</h3>`
Ich gehe davon aus, dass ich dort was ändern müsste, wenn ich anstelle des datestamps das angegebene Datum ausgegeben haben möchte.
--> z.B. http://optinomic.cust.local/api/#/patient/790/stay/471/app/ch.suedhang.apps.aase-g.production
weil ich die alten Fragebögen nacherfassen wollte, welche explizit nicht für Alkohol ausgefüllt wurden.

**Comment by schoenenb** [comment]
Gefunden in main.js :)

**Comment by ottigerb** [comment]
Der Zeitpunkt ist hier gesetzt:
https://github.com/Optinomic/apps/blob/master/ch.suedhang.apps.aase-g/main.js#L77
Akutell auf `response.datestamp`.

**Comment by schoenenb** [comment]
Um im BSCL die Sortierung nach Datum statt datestamp zu machen müsste ich 
https://github.com/Optinomic/apps/blob/master/ch.suedhang.apps.bscl_anq/main.js#L258 ff
anpassen?
Der Pfad sollte schlussendlich `sr.entity.data.response.q504V00` statt `sr.calculations["0"].calculation_result.info.filled` sein. Ich kann allerdings nicht nachvollziehen, was dort mit den Berechnungen noch passiert, die möchte ich ja schliesslich nicht beeinträchtigen.
Vielleicht hilft bereits ein Kaffee, ansonsten wäre ich froh um ein Eingreifen!

**Comment by ottigerb** [comment]
https://github.com/Optinomic/apps/blob/master/ch.suedhang.apps.bscl_anq/main.js#L265

Dies sollte problemlos sein:

```JS
current_calc.date = sr.entity.data.response.q504V00;
```

**Comment by schoenenb** [comment]
Fürs PDF habe ich die .filled-Zeilen ersetzt:
https://github.com/Optinomic/apps/blob/master/ch.suedhang.apps.pdf_druckvorlagen/scripts/app_bscl.js#L557
https://github.com/Optinomic/apps/blob/master/ch.suedhang.apps.pdf_druckvorlagen/scripts/app_bscl.js#L757
Die Resultate werden nun generell nach Eingabedatum sortiert, nur die Zusatzitems wollen noch nicht - ich komme leider nicht drauf, weshalb nicht.

**Comment by ottigerb** [comment]
... die sortieren nicht korrekt?

schoenenb <notifications@github.com> schrieb am Fr., 26. Mai 2017, 16:23:

**Comment by schoenenb** [comment]
Nein, nach wie vor umgekehrt bei http://optinomic.cust.local/api/#/patient/824/stay/588/app/ch.suedhang.apps.bscl_anq.production

**Comment by schoenenb** [comment]
Soviel ich sehe nicht:
http://optinomic.cust.local/api/#/patient/2940/stay/1363/report/ein____austritts_assessment

**Comment by schoenenb** [comment]
#### Angepasst sind die Daten bei folgenden Fragebögen:
|  | App | PDF |
| --- | --- | --- |
| BDI | Datum | Datum |
| SCI | Datum | Datum |
| BSCL Skalen | Datum | Datum |
| BSCL ZusItems| Datum | **filled** |

**Comment by schoenenb** [comment]
Ich wollte beim AASE eben gleich vorgehen, nur scheint die Reihenfolge weder in
https://github.com/Optinomic/apps/blob/master/ch.suedhang.apps.aase-g/main.js#L199 noch
https://github.com/Optinomic/apps/blob/master/ch.suedhang.apps.aase-g/main.js#L218
definiert zu sein - any hints?

**Comment by ottigerb** [comment]
AASE ::  `templates/score_scales.html` loopt über `ng-repeat="sr in d.dataMain.survey_responses"`. Entsprechend müssen wir survey_responses anpassen. Vgl. diesbezüglich:  53dd249, 9ea92a3 um das Datum zu setzen. Wichtig ist hier, dass nicht jede Messung ein 'Datum' in `response` hat - dann soll `filled` gezogen werden.  Zum Schluss können wir survey_responses sortieren https://github.com/Optinomic/apps/commit/805a975252810071178c0c12220218c8b3880b69. Hoffe es funktioniert wie erwartet.

**Comment by schoenenb** [comment]
Ahaa... Und wenn einmal nur filled vorhanden ist und einmal Datum, dann nimmt es filled - macht eigentlich Sinn. Das Problem ist nur entstanden, weil ich die früheren AASE nachtrage, welche nicht für Alkohol ausgefüllt wurden, wofür damals aber noch keine Unterscheidung vorhanden war...

:+1:  Funktioniert - http://optinomic.cust.local/api/#/patient/1164/stay/803/app/ch.suedhang.apps.aase-g.production :)

**Comment by schoenenb** [comment]
Dieses Issue ist glaube ich nur noch offen, wegen den BSCL Zusatzitems, bei welchen im PDF noch filled anstelle des Datums angegeben wurde - erledigt sich das mit der neuen PDF-Version?
Wenn ja - close, sonst gerne ein Auge drauf haben. :)
